lab_3_generate_by_ngrams/main.py

```diff
@@ -331,6 +520,20 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not (isinstance(sequence, tuple)
+                and isinstance(next_tokens, list)
+                and isinstance(sequence_candidates, dict) and sequence
+                and next_tokens and sequence_candidates and len(next_tokens) <= self._beam_width
+                and sequence in sequence_candidates
+        ):
+            return None
+
+        for token in next_tokens:
+            sequence_candidates[sequence + (token[0],)] = sequence_candidates[sequence] - math.log(token[1])
+
+        del sequence_candidates[sequence]
```

Comment:
Beware of mutation!!! The function should not change values outside of itself, yours does